README.md

```diff
@@ -174,34 +183,147 @@ Properties starting with a `$` are special properties. It must be one of the fol
+            <code>°</code> Degree<br>
+            <code>L</code> Liter<br>
+            <code>gal</code> Galon<br>
+            <code>V</code> Volts<br>
+            <code>W</code> Watt<br>
+            <code>A</code> Ampere<br>
+            <code>%</code> Percent<br>
+            <code>m</code> Meter<br>
+            <code>ft</code> Feet<br>
+            <code>Pa</code> Pascal<br>
+            <code>psi</code> PSI<br>
+            <code>#</code> Count or Amount
+        </td>
+        <td>Yes</td>
+        <td>Yes</td>
+    </tr>
+    <tr>
+       <td>$datatype</td>
+       <td>Device → Controller</td>
+       <td>Describes the format of data.</td>
+       <td><code>integer</code>, <code>float</code>, <code>boolean</code> (<code>true</code> or <code>false</code>), <code>string</code>, <code>enum</code> </td>
+       <td>Yes</td>
+       <td>Yes</td>
+    </tr>
+    <tr>
+       <td>$name</td>
+       <td>Device → Controller</td>
+       <td>Friendly name of the property.</td>
+       <td>Any String </td>
+       <td>Yes</td>
+       <td>Yes</td>
+    </tr>
+    <tr>
+       <td>$range</td>
+       <td>Device → Controller</td>
+       <td>
+        Describes what are valid values for this property.
+       </td>
+       <td>
+             <ul>
+                <li><code>from:to</code> Describes a range of values e.g. <code>10:15</code>. <br>Valid for datatypes <code>integer</code>, <code>float</code> </li>
+                <li><code>value,value,value</code> for enumerating all valid values. Escape <code>,</code> by using <code>,,</code>. e.g. <code>A,B,C</code> or <code>ON,OFF,PAUSE</code>. <br>Valid for datatypes <code>enum</code> </li>
+                <li><code>regex:/pattern/flags</code> to provide a regex that can be used to match the value. e.g. <code>regex:/[A-Z][0-9]+/g</code>. <br>Valid for datatype <code>string</code></li>
+            </ul>
+       </td>
+       <td>Yes</td>
+       <td>Yes</td>
+    </tr>
+
+
+</table>
+
+
 For example, our `686f6d6965` above would send:
 
 ```
 homie/686f6d6965/temperature/$type → temperature
 homie/686f6d6965/temperature/$properties → degrees,unit
-homie/686f6d6965/temperature/unit → c
+homie/686f6d6965/temperature/degrees/$settable → false
+homie/686f6d6965/temperature/degrees/$unit → C
+homie/686f6d6965/temperature/degrees/$datatype → float
+homie/686f6d6965/temperature/degrees/$range → -20.0:60
 homie/686f6d6965/temperature/degrees → 12.07
 
 homie/686f6d6965/humidity/$type → humidity
 homie/686f6d6965/humidity/$properties → percentage
+homie/686f6d6965/humidity/percentage/$settable → false
+homie/686f6d6965/humidity/percentage/$unit → %
+homie/686f6d6965/humidity/percentage/$datatype → integer
+homie/686f6d6965/humidity/percentage/$range → 0:100
 homie/686f6d6965/humidity/percentage → 79
 ```
 
 A LED strip would look like this. Note that the topic for a range properties is the name of the property followed by a `_` and the index getting updated:
 
```

Comment:
Agreed, this might get somehow confusing. I've changed the `$range` to `$format`, as "range" is the wrong term anyways. e.g. a regex is definitely not a "range", same for enums.